File: programovani/ulohy/files/pascaluv-trojuhelnik/pascaluv-trojuhelnik.py
# Deklarace a inicializace
i = 13 # Inicializujeme velikost trojuhelniku
pole = [] # Deklarujeme pole

# Cyklus pro radky
for j in range(i):

    # Deklarujeme a inicializujeme
    rad = [] # Prazdne pole, ktere pak budeme vkladat do promenne pole

    # Cyklus pro sloupce (jednotlive hodnoty) 
    for k in range(j + 1):

        # Pokud se jedna o prvni nebo posledni cislo
        if(k == 0 or k == j):

            # Vzdy je 1
            rad.append(1)

        # Pro ostatni cisla mezi prvnim a poslednim
        else:

            # Vlozime hodnoty podle vzorecku
            rad.append(pole[j-1][k-1] + pole[j-1][k])
    
    # Do vysledneho pole appendujeme (vlozime) celou radku
    pole.append(rad)

# Vypiseme Pascaluv trojuhelnik, cyklus pro radky
for j in range(i):

    # Cyklus pro sloupce (jednotlive hodnoty) 
    for k in range(len(pole[j])):

        # Vypiseme jednotlive hodnoty, ktere oddelime tabulatorem
        print(pole[j][k], end="\t")

    # Na konci vlozime novou radku
    print("")

File: programovani/ulohy/files/pascaluv-trojuhelnik/pascaluv-trojuhelnik.js
// Deklarace a inicializace
const i = 13; // Inicializujeme velikost trojuhelniku
var pole = []; // Deklarujeme pole

// Cyklus pro radky
for(var j = 0; j < i; j++) {

    // Do pole vlozime vlozime dalsi pole
    pole[j] = [];

    // Cyklus pro sloupce (jednotlive hodnoty)
    for(var k = 0; k <= j; k++) {

        // Pokud se jedna o prvni nebo posledni cislo
        if(k == 0 || k == j) {

            // Vzdy je 1
            pole[j][k] = 1;

        }

        // Pro ostatni cisla mezi prvnim a poslednim
        else {

            // Vlozime hodnoty podle vzorecku
            pole[j][k] = pole[j-1][k-1] + pole[j-1][k];

        }

    }

}

// Vypiseme Pascaluv trojuhelnik, cyklus pro radky
for(var j = 0; j < pole.length; j++) {

    // Deklarujeme a incializujeme
    var rad = ""; // Vypisovat musime radku jako text

    // Cyklus pro sloupce (jednotlive hodnoty) 
    for(var k = 0; k < pole[j].length; k++) {

        // Kazdou hodnotu pridame jako string do textu a oddelime tabulatorem
        rad += pole[j][k] + "\t";

    }

    // Vypiseme radku
    console.log(rad);
    
}